devito/symbolics/manipulation.py

```diff
@@ -319,12 +319,88 @@ def reuse_if_untouched(expr, args, evaluate=False):
         return expr.func(*args, evaluate=evaluate)
 
 
-def evalmin(a, b):
+def evalrel(func=min, input=None, assumptions=None):
     """
-    Simplify min(a, b) if possible.
+    The purpose of this function is two-fold: (i) to reduce the `input` candidates of a
+    for a MIN/MAX expression based on the given `assumptions` and (ii) return the nested
+    MIN/MAX expression of the reduced-size input. For this purpose this function takes
+    advantage of Python's built-in `min`, SymPy's `Min` and our in-house `MIN`.
+
+    Parameters
+    ----------
+    func : min or max, the operation to follow
```

Comment:
this is not valid numpydoc. IIRC the part after the ': ' is the data type. The description goes below

It applies to the other parameters as well